tor-button@shatur/files/tor-button@shatur/applet.js

```diff
@@ -0,0 +1,109 @@
+        this.settings.bindProperty(Settings.BindingDirection.IN,
+        "showNotifications",
+        "showNotifications",
+        this.on_settings_changed,
+        null);
+        
+        // Part of l10n support;
+        Gettext.bindtextdomain(UUID, GLib.get_home_dir() + "/.local/share/locale");
+        
+        //Icon and tooltip
+        this.set_applet_icon_symbolic_name('tor-off');
+        this.set_applet_tooltip(_(press_on));;
+        
+        //Create loop
+        Mainloop.timeout_add_seconds(this.refreshInterval, Lang.bind(this, this.updateLoop));
+    },
+    
+    // Inhibit the update timer when applet removed from panel
+    on_applet_removed_from_panel: function () {
+        applet_running = false;
+        this.settings.finalize();
+    },
+    
+    // Update loop timer and check status of Tor
+    updateLoop: function () {
+        if ( applet_running == true) {
+            GLib.spawn_command_line_async('sh ' + GLib.get_home_dir() + '/.local/share/cinnamon/applets/'+UUID+'/check.sh'); // Run check bash script to write Tor status in '/tmp/.torAppletCheck'
+            torAppletCheck = GLib.file_get_contents("/tmp/.torAppletCheck").toString(); // Read information from file
+            if ( torEnable != torAppletCheck ) // Check status change
+            {
+                if (torAppletCheck.substr(5,3) == "OFF") { // Check information from file without 'true,' and '\n '
+                                                           this.set_applet_icon_symbolic_name('tor-off'); // Change icon
+                                                           if (this.showNotifications == 1) { GLib.spawn_command_line_async('notify-send \'Tor\' \'Tor closed.\' --icon=dialog-information'); } // Show notification                 
+                                                           this.set_applet_tooltip(_(press_on)); // Change tooltip
+                                                           torEnable = torAppletCheck; // Set previous information for Tor status
+                }
+                else {
+                    this.set_applet_icon_symbolic_name('tor-on');
+                    if (this.showNotifications == 1) { GLib.spawn_command_line_async('notify-send \'Tor\' \'Tor launched.\' --icon=dialog-information'); }                    
+                    this.set_applet_tooltip(_(press_off));
+                    torEnable = torAppletCheck;
+                }
+            }
+            Mainloop.timeout_add_seconds(this.refreshInterval, Lang.bind(this, this.updateLoop)); // Update loop
+        }
+    },
+    
+    on_applet_clicked: function() {
+        if (torAppletCheck.substr(5,3) == "OFF") { // Check information from file without 'true,' and '\n '
+                                                   GLib.spawn_command_line_async(this.location); // Run tor from location settings        
+        }
+        else {
+            GLib.spawn_command_line_async('pkill tor');  // Kill Tor                 
```

Comment:
I'm launching Tor without su privilegies. But I add ability to run and kill Tor with gksu.
Killing Tor is needed when you want to disable Tor. I not found how to stop service and decided that I can just kill service.